### PR TITLE
Story/fire starts role/1869

### PR DESCRIPTION
**Pull request by conbrad**
- Adds backend logic to 401 any request from a token that does not have `hfi_set_fire_starts` role assigned
- Adds frontend logic to disable fire starts dropdowns if token does not have `hfi_set_fire_starts` role assigned
- Simplifies `AuthWrapper` component (removes `shouldAuthenticate` boolean, just checks for `window.Cypress or TEST_AUTH env variable)
- Adds tests around authentication redux store
# Test Links:
[Percentile Calculator](https://wps-pr-1890.apps.silver.devops.gov.bc.ca/)
[MoreCast](https://wps-pr-1890.apps.silver.devops.gov.bc.ca/morecast)
[C-Haines](https://wps-pr-1890.apps.silver.devops.gov.bc.ca/c-haines)
[FireBat](https://wps-pr-1890.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator)
[FireBat bookmark](https://wps-pr-1890.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator?s=266&f=c5&c=NaN&w=20,s=286&f=c7&c=NaN&w=16,s=1055&f=c7&c=NaN&w=NaN,s=305&f=c7&c=NaN&w=NaN,s=344&f=c5&c=NaN&w=NaN,s=346&f=c7&c=NaN&w=NaN,s=328&f=c7&c=NaN&w=NaN,s=1399&f=c7&c=NaN&w=NaN,s=334&f=c7&c=NaN&w=NaN,s=1082&f=c3&c=NaN&w=NaN,s=388&f=c7&c=NaN&w=NaN,s=309&f=c7&c=NaN&w=16,s=306&f=c7&c=NaN&w=NaN,s=1029&f=c7&c=NaN&w=NaN,s=298&f=c7&c=NaN&w=NaN,s=1108&f=c5&c=NaN&w=NaN,s=836&f=c7&c=NaN&w=NaN#state=2ec784ca-c46a-49d0-b2b3-1cf32a9015a2&session_state=7d9447c8-db66-4661-b4cb-03d2ac0d1d8f&code=32292df4-2bdf-4f90-a4a8-c8dbcda682a9.7d9447c8-db66-4661-b4cb-03d2ac0d1d8f.2b63f390-f3dc-43ae-89f2-016453863476)
[Fire Behaviour Advisory](https://wps-pr-1890.apps.silver.devops.gov.bc.ca/fire-behaviour-advisory)
[HFI Calculator](https://wps-pr-1890.apps.silver.devops.gov.bc.ca/hfi-calculator)
[FWI Calculator](https://wps-pr-1890.apps.silver.devops.gov.bc.ca/fwi-calculator)

### CODE REVIEW

File: web/src/features/auth/AuthWrapper.tsx [hunk 1]
```diff
@@ -1,15 +1,19 @@
...
 import { AppThunk } from 'app/store'
 import { selectToken, selectAuthentication } from 'app/rootReducer'
+import { TEST_AUTH } from 'utils/env'
 
 interface Props {
-  shouldAuthenticate: boolean
   children: React.ReactElement
 }
 
+// Used for test authentication
```
Comment (Sybrand) [inline web/src/features/auth/AuthWrapper.tsx:13]: A final thought (repeat thought?) - we could load the token from environment variables. That way we're a) not throwing your token out to the world b) not storing dumb test stuff in our main code.

File: web/src/features/auth/roles.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+export const ROLES = {
+  HFI: {
+    SET_FIRE_STARTS: 'hfi_set_fire_starts'
```
Comment (Sybrand) [inline web/src/features/auth/roles.ts:3]: I guess we could load this from the environment too - but that seems excessive.

File: web/src/features/auth/slices/authenticationSlice.test.ts [hunk 3]
```diff
@@ -1,18 +1,105 @@
...
+  decodeRoles,
+  initialState,
+  authenticateStart,
+  authenticateFinished,
+  authenticateError,
+  refreshTokenFinished
+} from 'features/auth/slices/authenticationSlice'
 
 describe('authenticationSlice', () => {
+  const tokenWithRoles =
```
Comment (Sybrand) [inline web/src/features/auth/slices/authenticationSlice.test.ts:11]: I don't mind this in the tests - but feels wrong having something like this in web/src/features/auth/AuthWrapper.tsx

File: web/src/features/auth/slices/authenticationSlice.ts [hunk 4]
```diff
@@ -86,6 +87,12 @@ export const decodeRoles = (token: string | undefined) => {
   }
 }
 
+export const testAuthenticate =
```
Comment (Sybrand) [inline web/src/features/auth/slices/authenticationSlice.ts:90]: Mentioned this before - but maybe a better name? My mind immediately starts expecting that I'm in a test file.

File: api/app/tests/hfi/test_hfi_endpoint_request.py [hunk 5]
```diff
@@ -11,7 +11,7 @@
 import app.routers.hfi_calc
```
Comment (Sybrand) [inline api/app/tests/hfi/test_hfi_endpoint_request.py:11]: Dumb question - and maybe not part of this PR - but are we testing anywhere that an expired token kicks you out on the back-end?

**Review by Sybrand — APPROVED** [review]
Pls. look at comments.

File: web/src/features/auth/roles.ts (hunk 2 above)
Comment (conbrad) [reply web/src/features/auth/roles.ts:3]: Feels excessive to me -- I also feel our set of environment variables is already excessive and unwieldy.

Comment (Sybrand) [reply web/src/features/auth/roles.ts:3]: Agreed

File: web/src/features/auth/AuthWrapper.tsx (hunk 1 above)
Comment (Sybrand) [reply web/src/features/auth/AuthWrapper.tsx:13]: It DOES mean that depending on your configuration you'll get different test results. That's a bad.....

File: api/app/tests/hfi/test_hfi_endpoint_request.py (hunk 5 above)
Comment (conbrad) [reply api/app/tests/hfi/test_hfi_endpoint_request.py:11]: We have these tests on the backend: https://github.com/bcgov/wps/blob/main/api/app/tests/test_auth.feature

File: web/src/features/auth/slices/authenticationSlice.test.ts (hunk 3 above)
Comment (conbrad) [reply web/src/features/auth/slices/authenticationSlice.test.ts:11]: Not sure where to put the one in `AuthWrapper.tsx` besides maybe `web/src/features/auth/roles.ts` only because it's a set of constants in there to do with roles/auth.
